core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala

```diff
@@ -194,4 +194,42 @@ class OpenHashMapSuite extends SparkFunSuite with Matchers {
     val numInvalidValues = map.iterator.count(_._2 == 0)
     assertResult(0)(numInvalidValues)
   }
+
+  test("distinguish between the 0/0.0/0L and null") {
+    val specializedMap1 = new OpenHashMap[String, Long]
+    specializedMap1("a") = null.asInstanceOf[Long]
+    specializedMap1("b") = 0L
+    assert(specializedMap1.contains("a"))
+    assert(!specializedMap1.contains("c"))
+    assert(Some(specializedMap1("a")).contains(0L))
+    assert(Some(specializedMap1("b")).contains(0L))
+    assert(Some(specializedMap1("c")).contains(0L))
```

Comment:
What do these tests mean? Tests for different type comparisons? e.g.,
```
    assert(!Some(specializedMap1("a")).contains(0.0))
    assert(!Some(specializedMap1("b")).contains(0.toShort))
```

Reply:
if the data type is in `@specialized annotation`  , and the `key` not be contained ,the `map(key)` will  return 0
if the data type is not  in `@specialized annotation`  , and the `key` not be contained ,the `map(key)` will  return null

Reply:
ok , i will add it